modules/extension/wms/src/main/java/org/geotools/ows/wms/xml/Dimension.java

```diff
@@ -51,7 +51,8 @@ public Dimension(String name, String units, String unitSymbol) {
         if (name == null || name.length() == 0) {
             throw new IllegalArgumentException("Error creating Extent: parameter name must not be null!");
         }
-        if (units == null || units.length() == 0) {
+        // unit must not be null, but can be empty
+        if (units == null) {
```

Comment:
This is a bizzarre choice in the spec, I would quote the relevant spec section in the class javadoc for clarification (to clarify the code is not just leveraging a loophole in the schema, it was the actual specification intention):

> * If the dimensional quantity has no units (e.g. band number in a multi-wavelength sensor), use the null string: units="".